### ISSUE TITLE
Dome light on near-horizontal surfaces and texture wrapping

### ISSUE THREAD

**Issue by tdelame** [PixarAnimationStudios/USD#1391]
### Description of Issue
Dome light on near-horizontal surfaces create some artefacts because:
- GetTexCoords() in `domeLight.glslfx` should apply an offset of (0.5, 0.5) to compute samples at pixel centers. Otherwise, samples are computed at pixel corners and the bottom line of the irradiance map is incorrect (see below):
![buggy_irradiance](https://user-images.githubusercontent.com/12368589/99086037-4454e280-25c9-11eb-89ed-2c78ab25e231.png)
Applying the offset produces the following irradiance map:
![fixed_irradiance](https://user-images.githubusercontent.com/12368589/99086308-a01f6b80-25c9-11eb-914c-17587d5cb16e.png)
- texture samplers should have HdWrapClamp mode for T and not HdWrapRepeat. Otherwise when the surface is near-horizontal, there can be a discontinuity producing this image
![buggy_render](https://user-images.githubusercontent.com/12368589/99086483-e2e14380-25c9-11eb-87dc-638db65acf7d.png)
instead of this one
![fixed_render](https://user-images.githubusercontent.com/12368589/99086641-158b3c00-25ca-11eb-87ea-81c52737a558.png)
- normals should be clamped in [-1,1] after being transformed to world coordinates before fetching texels in dome light textures, otherwise there can be artefacts on near-horizontal surfaces depending on the view point

### Steps to Reproduce
1. Open the following file (a floor tile from the Kitchen set with a UsdPreviewSurface shader applied to it) and render it
```
#usda 1.0

def Xform "root" {
  matrix4d xformOp:transform = ( (1, 0, 0, 0), (0, 1, 0, 0), (0, 0, 1, 0), (0, 0, 0, 1) )
  uniform token[] xformOpOrder = ["xformOp:transform"]

  def Mesh "object" {
    float3[] extent = [(-9.618811, -16.597124, -0.9819515), (24.007746, 17.029436, 0.6106356)]
    int[] faceVertexCounts = [4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4]
    int[] faceVertexIndices = [8, 12, 25, 18, 0, 20, 21, 16, 0, 16, 70, 71, 1, 22, 79, 68, 15, 2, 74, 75, 20, 0, 71, 72, 4, 17, 46, 47, 5, 23, 55, 44, 14, 7, 50, 51, 19, 4, 47, 48, 12, 41, 42, 25, 35, 14, 51, 52, 36, 15, 75, 76, 16, 21, 38, 39, 70, 16, 39, 69, 46, 17, 40, 45, 27, 19, 48, 49, 28, 20, 72, 73, 21, 20, 28, 29, 38, 21, 29, 37, 79, 22, 30, 78, 55, 23, 31, 54, 25, 42, 43, 33, 18, 25, 33, 26, 7, 27, 49, 50, 2, 28, 73, 74, 29, 28, 2, 15, 37, 29, 15, 36, 78, 30, 3, 77, 54, 31, 6, 53, 33, 43, 34, 13, 26, 33, 13, 11, 6, 35, 52, 53, 3, 36, 76, 77, 30, 37, 36, 3, 22, 38, 37, 30, 39, 38, 22, 1, 69, 39, 1, 68, 45, 40, 5, 44, 42, 41, 9, 24, 43, 42, 24, 32, 34, 43, 32, 10, 41, 45, 44, 9, 12, 46, 45, 41, 47, 46, 12, 8, 48, 47, 8, 18, 49, 48, 18, 26, 50, 49, 26, 11, 51, 50, 11, 13, 52, 51, 13, 34, 53, 52, 34, 10, 32, 54, 53, 10, 24, 55, 54, 32, 44, 55, 24, 9, 40, 57, 56, 5, 17, 58, 57, 40, 59, 58, 17, 4, 60, 59, 4, 19, 61, 60, 19, 27, 62, 61, 27, 7, 63, 62, 7, 14, 64, 63, 14, 35, 65, 64, 35, 6, 31, 66, 65, 6, 23, 67, 66, 31, 56, 67, 23, 5, 57, 69, 68, 56, 58, 70, 69, 57, 71, 70, 58, 59, 72, 71, 59, 60, 73, 72, 60, 61, 74, 73, 61, 62, 75, 74, 62, 63, 76, 75, 63, 64, 77, 76, 64, 65, 66, 78, 77, 65, 67, 79, 78, 66, 68, 79, 67, 56]
    point3f[] points = [(-9.618811, -16.597124, -0.9819515), (24.007746, -16.597124, -0.98152447), (-9.618811, 17.029436, -0.98195046), (24.007746, 17.029436, -0.98195046), (-9.618811, -16.597124, -0.28416878), (24.007746, -16.597124, 0.10466725), (24.007746, 17.029434, -0.28416878), (-9.618811, 17.029434, -0.28416878), (-8.165766, -15.144078, 0.0408704), (22.554703, -15.144078, 0.6106356), (22.554703, 15.576391, 0.04087037), (-8.165766, 15.576391, 0.04087037), (-7.3563905, -15.144078, 0.0408704), (-7.3563905, 15.576391, 0.04087037), (-8.73287, 17.029434, -0.28416878), (-8.73287, 17.029436, -0.98195046), (-8.73287, -16.597124, -0.9819515), (-8.73287, -16.597124, -0.28416878), (-8.165766, -14.080504, 0.0408704), (-9.618811, -15.432938, -0.28416878), (-9.618811, -15.432938, -0.9819505), (-8.73287, -15.432938, -0.9819505), (24.007746, -15.432938, -0.98152554), (24.007746, -15.432938, 0.10466725), (22.554703, -14.080504, 0.6106356), (-7.3563895, -14.080504, 0.0408704), (-8.165766, 15.001377, 0.0408704), (-9.618811, 16.400024, -0.28416878), (-9.618811, 16.400024, -0.98195046), (-8.73287, 16.400024, -0.98195046), (24.007746, 16.400024, -0.98195046), (24.007746, 16.400024, -0.28416878), (22.554703, 15.001377, 0.0408704), (-7.3563905, 15.001377, 0.0408704), (21.849794, 15.576391, 0.04087037), (23.236156, 17.029434, -0.28416878), (23.236156, 17.029436, -0.98195046), (23.236156, 16.400024, -0.98195046), (23.236156, -15.432938, -0.98152554), (23.236156, -16.597124, -0.98152447), (23.236156, -16.597124, 0.10466725), (21.849794, -15.144078, 0.6106356), (21.849794, -14.080504, 0.6106356), (21.849794, 15.001377, 0.0408704), (23.417551, -16.006926, 0.56415784), (22.673044, -16.006926, 0.56415784), (-8.173773, -16.006926, 0.011012375), (-9.028614, -16.006926, 0.011012375), (-9.028614, -14.883606, 0.011012375), (-9.028614, 15.831925, 0.011012346), (-9.028614, 16.439238, 0.011012346), (-8.173773, 16.439238, 0.011012346), (22.673044, 16.439238, 0.011012346), (23.417551, 16.439238, 0.011012346), (23.417551, 15.831925, 0.011012346), (23.417551, -14.883606, 0.5641578), (24.007746, -16.597122, -0.00044620037), (23.236156, -16.597122, -0.00044620037), (-8.73287, -16.597122, -0.35169458), (-9.618811, -16.597122, -0.35169458), (-9.618811, -15.432938, -0.35169458), (-9.618811, 16.400026, -0.35169464), (-9.618811, 17.029434, -0.3516956), (-8.73287, 17.029434, -0.3516956), (23.236156, 17.029434, -0.3516956), (24.007746, 17.029434, -0.3516956), (24.007746, 16.400026, -0.35169464), (24.007746, -15.432938, -0.00044620037), (24.007746, -16.597124, -0.8352587), (23.236156, -16.597124, -0.8352587), (-8.73287, -16.597124, -0.88798785), (-9.618811, -16.597124, -0.88798785), (-9.618811, -15.432938, -0.88798785), (-9.618811, 16.400024, -0.8879879), (-9.618811, 17.029436, -0.8879879), (-8.73287, 17.029436, -0.8879879), (23.236156, 17.029436, -0.8879879), (24.007746, 17.029436, -0.8879879), (24.007746, 16.400024, -0.8879879), (24.007746, -15.432938, -0.8352599)]
    float[] primvars:__faceindex = [0, 0, 0, 0, 1, 1, 1, 1, 2, 2, 2, 2, 3, 3, 3, 3, 4, 4, 4, 4, 5, 5, 5, 5, 6, 6, 6, 6, 7, 7, 7, 7, 8, 8, 8, 8, 9, 9, 9, 9, 10, 10, 10, 10, 11, 11, 11, 11, 12, 12, 12, 12, 13, 13, 13, 13, 14, 14, 14, 14, 15, 15, 15, 15, 16, 16, 16, 16, 17, 17, 17, 17, 18, 18, 18, 18, 19, 19, 19, 19, 20, 20, 20, 20, 21, 21, 21, 21, 22, 22, 22, 22, 23, 23, 23, 23, 24, 24, 24, 24, 25, 25, 25, 25, 26, 26, 26, 26, 27, 27, 27, 27, 28, 28, 28, 28, 29, 29, 29, 29, 30, 30, 30, 30, 31, 31, 31, 31, 32, 32, 32, 32, 33, 33, 33, 33, 34, 34, 34, 34, 35, 35, 35, 35, 36, 36, 36, 36, 37, 37, 37, 37, 38, 38, 38, 38, 39, 39, 39, 39, 40, 40, 40, 40, 41, 41, 41, 41, 42, 42, 42, 42, 43, 43, 43, 43, 44, 44, 44, 44, 45, 45, 45, 45, 46, 46, 46, 46, 47, 47, 47, 47, 48, 48, 48, 48, 49, 49, 49, 49, 50, 50, 50, 50, 51, 51, 51, 51, 52, 52, 52, 52, 53, 53, 53, 53, 54, 54, 54, 54, 55, 55, 55, 55, 56, 56, 56, 56, 57, 57, 57, 57, 58, 58, 58, 58, 59, 59, 59, 59, 60, 60, 60, 60, 61, 61, 61, 61, 62, 62, 62, 62, 63, 63, 63, 63, 64, 64, 64, 64, 65, 65, 65, 65, 66, 66, 66, 66, 67, 67, 67, 67, 68, 68, 68, 68, 69, 69, 69, 69, 70, 70, 70, 70, 71, 71, 71, 71, 72, 72, 72, 72, 73, 73, 73, 73, 74, 74, 74, 74, 75, 75, 75, 75, 76, 76, 76, 76, 77, 77, 77, 77] (
        interpolation = "faceVarying"
    )
    color3f[] primvars:displayColor = [(0.47843137, 0.4627451, 0.42745098)]
    float3[] primvars:PreMenvPosingRefPose = [(-9.618811, -16.597124, -0.9819515), (24.007746, -16.597124, -0.98152447), (-9.618811, 17.029436, -0.98195046), (24.007746, 17.029436, -0.98195046), (-9.618811, -16.597124, -0.28416878), (24.007746, -16.597124, 0.10466725), (24.007746, 17.029434, -0.28416878), (-9.618811, 17.029434, -0.28416878), (-8.165766, -15.144078, 0.0408704), (22.554703, -15.144078, 0.6106356), (22.554703, 15.576391, 0.04087037), (-8.165766, 15.576391, 0.04087037), (-7.3563905, -15.144078, 0.0408704), (-7.3563905, 15.576391, 0.04087037), (-8.73287, 17.029434, -0.28416878), (-8.73287, 17.029436, -0.98195046), (-8.73287, -16.597124, -0.9819515), (-8.73287, -16.597124, -0.28416878), (-8.165766, -14.080504, 0.0408704), (-9.618811, -15.432938, -0.28416878), (-9.618811, -15.432938, -0.9819505), (-8.73287, -15.432938, -0.9819505), (24.007746, -15.432938, -0.98152554), (24.007746, -15.432938, 0.10466725), (22.554703, -14.080504, 0.6106356), (-7.3563895, -14.080504, 0.0408704), (-8.165766, 15.001377, 0.0408704), (-9.618811, 16.400024, -0.28416878), (-9.618811, 16.400024, -0.98195046), (-8.73287, 16.400024, -0.98195046), (24.007746, 16.400024, -0.98195046), (24.007746, 16.400024, -0.28416878), (22.554703, 15.001377, 0.0408704), (-7.3563905, 15.001377, 0.0408704), (21.849794, 15.576391, 0.04087037), (23.236156, 17.029434, -0.28416878), (23.236156, 17.029436, -0.98195046), (23.236156, 16.400024, -0.98195046), (23.236156, -15.432938, -0.98152554), (23.236156, -16.597124, -0.98152447), (23.236156, -16.597124, 0.10466725), (21.849794, -15.144078, 0.6106356), (21.849794, -14.080504, 0.6106356), (21.849794, 15.001377, 0.0408704), (23.417551, -16.006926, 0.56415784), (22.673044, -16.006926, 0.56415784), (-8.173773, -16.006926, 0.011012375), (-9.028614, -16.006926, 0.011012375), (-9.028614, -14.883606, 0.011012375), (-9.028614, 15.831925, 0.011012346), (-9.028614, 16.439238, 0.011012346), (-8.173773, 16.439238, 0.011012346), (22.673044, 16.439238, 0.011012346), (23.417551, 16.439238, 0.011012346), (23.417551, 15.831925, 0.011012346), (23.417551, -14.883606, 0.5641578), (24.007746, -16.597122, -0.00044620037), (23.236156, -16.597122, -0.00044620037), (-8.73287, -16.597122, -0.35169458), (-9.618811, -16.597122, -0.35169458), (-9.618811, -15.432938, -0.35169458), (-9.618811, 16.400026, -0.35169464), (-9.618811, 17.029434, -0.3516956), (-8.73287, 17.029434, -0.3516956), (23.236156, 17.029434, -0.3516956), (24.007746, 17.029434, -0.3516956), (24.007746, 16.400026, -0.35169464), (24.007746, -15.432938, -0.00044620037), (24.007746, -16.597124, -0.8352587), (23.236156, -16.597124, -0.8352587), (-8.73287, -16.597124, -0.88798785), (-9.618811, -16.597124, -0.88798785), (-9.618811, -15.432938, -0.88798785), (-9.618811, 16.400024, -0.8879879), (-9.618811, 17.029436, -0.8879879), (-8.73287, 17.029436, -0.8879879), (23.236156, 17.029436, -0.8879879), (24.007746, 17.029436, -0.8879879), (24.007746, 16.400024, -0.8879879), (24.007746, -15.432938, -0.8352599)] (
        interpolation = "vertex"
    )
    float2[] primvars:st = [(0, 0), (0.02634647, 0), (0.02634647, 0.03462098), (0, 0.03462098), (0.02634647, 0.034621), (1, 0), (1, 0.03462098), (0.02634647, 1), (0, 1), (0.9770541, 0), (0.9770541, 0.03462098), (0.9770541, 1), (0, 0.98128235), (0.02634647, 0.98128235), (0.9770541, 0.98128235), (1, 0.98128235), (1, 1)] (
        interpolation = "vertex"
    )
    int[] primvars:st:indices = [0, 5, 8, 16, 0, 5, 16, 8, 0, 5, 16, 8, 1, 7, 7, 7, 1, 1, 3, 3, 3, 4, 6, 6, 6, 2, 12, 12, 12, 13, 15, 15, 15, 13, 11, 11, 11, 14, 10, 9, 9, 9, 10, 14, 5, 9, 1, 0, 3, 12, 8, 7, 11, 16, 15, 6, 5, 9, 1, 0, 3, 12, 8, 7, 11, 16, 15, 6, 5, 9, 1, 0, 3, 12, 8, 7, 11, 16, 15, 6]
    rel material:binding = </materials/material_0>
  }
}

def "materials"
{
    def Material "material_0"
    {
        token outputs:surface.connect = </materials/material_0/shader.outputs:surface>

        def Shader "shader"
        {
            uniform token info:id = "UsdPreviewSurface"
            float3 inputs:diffuseColor = (1, 1, 1)
            float inputs:metallic = 0.1
            float inputs:opacity = 1
            float inputs:roughness = 0.1
            token outputs:surface
        }
    }
}
```

### Package Versions
Usd 20.11

**Comment by jilliene** [comment]
Filed as internal issue #USD-6482

**Comment by tallytalwar** [comment]
Thanks, included in the 21.02 release.
